client/modules/IDE/reducers/search.js

```diff
@@ -1,15 +1,24 @@
-import * as ActionTypes from '../../../constants';
+import { createSlice } from '@reduxjs/toolkit';
 
 const initialState = {
   collectionSearchTerm: '',
   sketchSearchTerm: ''
 };
 
-export default (state = initialState, action) => {
-  switch (action.type) {
-    case ActionTypes.SET_SEARCH_TERM:
-      return { ...state, [`${action.scope}SearchTerm`]: action.query };
-    default:
-      return state;
+const searchSlice = createSlice({
+  name: 'search',
+  initialState,
+  reducers: {
+    setSearchTerm: (state, action) => {
+      const { scope, query } = action.payload;
+      return {
+        ...state,
+        [`${scope}SearchTerm`]: query
+      };
```

Comment:
You don't need to do `...state` stuff when using Redux Toolkit.  The reducer uses a proxy of the actual state so it is okay to mutate it.  You can `return` a new state to replace the entire state, but typically you just modify the state and don't `return` anything.  It can be written as:
```
setSearchTerm: (state, action) => {
  const { scope, query } = action.payload;
  state[`${scope}SearchTerm`] = query;
}
```